daemon/src/taker.rs

```diff
@@ -279,9 +281,20 @@ async fn main() -> Result<()> {
         .mount(
             "/",
             rocket::routes![routes_taker::dist, routes_taker::index],
-        )
-        .launch()
-        .await?;
+        );
+
+    let rocket = rocket.ignite().await?;
+    let shutdown_handle = rocket.shutdown();
+    rocket::tokio::spawn(rocket.launch());
+
+    // shutdown the rocket server maker if goes offline
+    tokio::spawn(async move {
+        maker_online_status_feed_receiver.changed().await.unwrap();
```

Comment:
Under which circumstances can this `unwrap` fail ?

Reply:
"This method returns an error if and only if the Sender is dropped." - the sender will outlive the request, as it's managed. shall I use `.expect()` with some explanation instead?